File: java/src/main/java/hubspot2s3/S3.java
package hubspot2s3;

import java.io.InputStream;
import java.util.logging.Logger;
import com.amazonaws.AmazonServiceException;
import com.amazonaws.auth.AWSStaticCredentialsProvider;
import com.amazonaws.auth.BasicAWSCredentials;
import com.amazonaws.services.s3.AmazonS3;
import com.amazonaws.services.s3.AmazonS3ClientBuilder;
import com.amazonaws.services.s3.model.S3Object;
import com.amazonaws.services.s3.model.S3ObjectInputStream;

public class S3 {

	private static Logger logger = Logger.getLogger(Main.class.getName());
	private S3ObjectInputStream s3InputStream = null;

	S3(String accessKeyId, String secretAccessKey, String bucketName, String fileName, String region) {
		logger.info(" ----------------------------------- S3 -----------------------------------");
		logger.info("Downloading feed.");
		BasicAWSCredentials awsCreds = new BasicAWSCredentials(accessKeyId, secretAccessKey);
		AmazonS3 s3Client = AmazonS3ClientBuilder.standard().withCredentials(new AWSStaticCredentialsProvider(awsCreds))
				.withRegion(region).build();

		try {
			S3Object o = s3Client.getObject(bucketName, fileName);
			s3InputStream = o.getObjectContent();

		} catch (AmazonServiceException e) {
			System.err.println(e.getErrorMessage());
			System.exit(1);
		}
	}

	public InputStream getInputStream() {
		return s3InputStream;
	}
}

File: HUBSPOT_TO_S3/csharp/hubspot2s3/S3.cs
using System.IO;
using Amazon;
using Amazon.S3;
using Amazon.S3.Model;

namespace hubspot2s3
{
    public class S3
    {
        private readonly Stream _s3InputStream;
        private static readonly NLog.Logger s_logger = NLog.LogManager.GetCurrentClassLogger();

        public S3(string AccessKeyId, string SecretAccessKey, string BucketName, string FileName, string Region)
        {
            s_logger.Info(" ----------------------------------- S3 -----------------------------------");
            s_logger.Info("Downloading feed.");
            AmazonS3Client s3Client = new AmazonS3Client(AccessKeyId, SecretAccessKey, RegionEndpoint.GetBySystemName(Region));

            var _request = new GetObjectRequest
            {
                BucketName = BucketName,
                Key = FileName
            };

            using var getObjectResponse = s3Client.GetObjectAsync(_request);
            _s3InputStream = getObjectResponse.Result.ResponseStream;
        }

        public Stream S3InputStream
        {
            get { return _s3InputStream; }
        }
    }
}